app/ide-desktop/lib/performance-logger/src/performance_logger.ts

```diff
@@ -0,0 +1,245 @@
+export enum LogLevel {
+    Section = 'SECTION',
+    Task = 'TASK',
+    Detail = 'DETAIL',
+    Debug = 'DEBUG',
+}
+
+/// Delimiter used to to encode information in the `PerformanceEntry` name.
+const MESSAGE_DELIMITER = '//'
+
+const ENABLE_SECTION_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('section')
+const ENABLE_TASK_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('task')
+const ENABLE_DETAIL_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('detail')
+const ENABLE_DEBUG_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('debug')
+
+/// Check whether the given log level should perform any logging activity.
+function log_level_is_active(log_level: LogLevel): boolean {
+    switch (log_level) {
+        case LogLevel.Section:
+            return ENABLE_SECTION_PROFILING
+        case LogLevel.Task:
+            return ENABLE_TASK_PROFILING
+        case LogLevel.Detail:
+            return ENABLE_DETAIL_PROFILING
+        case LogLevel.Debug:
+            return ENABLE_DEBUG_PROFILING
+    }
```

Comment:
I think a lot of names are con fusing here. We are not logging anything with this framework (https://en.wikipedia.org/wiki/Logging_(software)). We are not keeping list of messages in this framework. This is a profiling framework, not a logging one.

Reply:
Also, this is quite a repetitive code. You can get the name of constructor here and make this code non-repetitive.

Reply:
Renamed and refactored.